### PR TITLE
feat(UIS-12) Add new item to inventory

### PR DESCRIPTION
**Pull request by StaceyJin**
JIRA: https://champlainsaintlambert.atlassian.net/browse/UIS-12

Add new item to inventory functionality was implemented to BACKEND ONLY:

method added to InventoryItemService and InventoryItemServiceImpl;
controller with method added;
relevant unit tests added.

### CODE REVIEW

File: src/main/java/com/um/inventoryservice/BusinessLayer/InventoryItemServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+public class InventoryItemServiceImpl implements InventoryItemService{
+
+    @Autowired
+    InventoryItemRepository inventoryItemRepository;
+
+    @Override
+    public Mono<InventoryItemDTO> insertInventoryItem(Mono<InventoryItemDTO> inventoryItemDTOMono) {
+        return inventoryItemDTOMono
+                .map(EntityDTOUtil::toEntity)
+//                .doOnNext(e -> e.(EntityDtoUtil.generateInventoryId()))
```
Comment (Kam1st) [inline src/main/java/com/um/inventoryservice/BusinessLayer/InventoryItemServiceImpl.java:20]: can take off the comment next sprint if we feel like we don't need to generate the ID anymore

File: src/main/java/com/um/inventoryservice/util/EntityDTOUtil.java [hunk 2]
```diff
@@ -14,23 +16,43 @@ public EntityDTOUtil(){}
...
         stockItem.setSalesQuantity(dto.getSalesQuantity());
         return stockItem;
     }
 
-    public static String generateStockId() {
-        Random random = new Random();
-        int number = random.nextInt(99999);
-        return "22" + (String.format("%05d", number));
-    }
+//    public static String generateStockId() {
```
Comment (Kam1st) [inline src/main/java/com/um/inventoryservice/util/EntityDTOUtil.java:27]: we can take it out next sprint if no need

File: src/main/java/com/um/inventoryservice/util/EntityDTOUtil.java [hunk 3]
```diff
@@ -14,23 +16,43 @@ public EntityDTOUtil(){}
...
+        return inventoryItem;
+    }
+
+    public static InventoryItemDTO toDTO(InventoryItem inventoryItem) {
+        InventoryItemDTO inventoryItemDTO = new InventoryItemDTO();
+        BeanUtils.copyProperties(inventoryItem, inventoryItemDTO);
+        return inventoryItemDTO;
+    }
+
+//    public static String generateInventoryId() {
```
Comment (Kam1st) [inline src/main/java/com/um/inventoryservice/util/EntityDTOUtil.java:53]: same as before

File: src/test/java/com/um/inventoryservice/PresentationLayer/InventoryControllerIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                .body(Mono.just(inventoryItem), InventoryItemDTO.class)
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBody(InventoryItemDTO.class)
+                .value((dto) -> {
+                    assertEquals(dto.getInventoryItemId(), inventoryItem.getInventoryItemId());
+                    assertEquals(dto.getQuantityInStock(), inventoryItem.getQuantityInStock());
+//                    assertThat(dto.getQuantityInStock()).isEqualTo(inventoryItem.getQuantityInStock());
```
Comment (Kam1st) [inline src/test/java/com/um/inventoryservice/PresentationLayer/InventoryControllerIntegrationTest.java:54]: remove this line next merge

**Review by Kam1st — APPROVED** [review]

File: src/test/java/com/um/inventoryservice/PresentationLayer/InventoryControllerIntegrationTest.java (hunk 4 above)
Comment (StaceyJin) [reply src/test/java/com/um/inventoryservice/PresentationLayer/InventoryControllerIntegrationTest.java:54]: Already removed.

**Pull request merged**